taskiq_redis/taskiq/brokers/redis/redis_broker.py

```diff
@@ -0,0 +1,106 @@
+
+from redis.asyncio import ConnectionPool, Redis
+from taskiq.abc.broker import AsyncBroker
+from taskiq.abc.result_backend import AsyncResultBackend
+from taskiq.message import BrokerMessage
+
+_T = TypeVar("_T")  # noqa: WPS111
+
+logger = getLogger("taskiq.redis_broker")
+
+
+class RedisBroker(AsyncBroker):
+    """Broker that works with Redis."""
+
+    def __init__(
+        self,
+        url: Optional[str] = None,
+        task_id_generator: Optional[Callable[[], str]] = None,
+        result_backend: Optional[AsyncResultBackend[_T]] = None,
+        queue_name: str = "taskiq",
+        max_connection_pool_size: Optional[int] = None,
+        **connection_kwargs: Any,
+    ) -> None:
+        """
+        Constructs a new broker.
+
+        :param url: url to redis.
+        :param task_id_generator: custom task_id generator.
+        :param result_backend: custom result backend.
+        :param queue_name: name for a list in redis.
+        :param max_connection_pool_size: maximum number of connections in pool.
+        :param connection_kwargs: additional arguments for aio-redis ConnectionPool.
+        """
+        super().__init__(
+            result_backend=result_backend,
+            task_id_generator=task_id_generator,
+        )
+
+        self.connection_pool: ConnectionPool = ConnectionPool.from_url(
+            url=url,
+            max_connections=max_connection_pool_size,
+            **connection_kwargs,
+        )
+
+        self.redis_list_name = queue_name
+
+    async def shutdown(self) -> None:
+        """Closes redis connection pool."""
+        self.connection_pool.disconnect()
+
+    async def kick(self, message: BrokerMessage) -> None:
+        """
+        Sends a message to the redis broker list.
+
+        This function constructs message for redis
+        and sends it.
+
+        The message is pickled dict object with message,
+        task_id, task_name and labels.
+
+        :param message: message to send.
+        """
+        async with Redis(connection_pool=self.connection_pool) as redis_conn:
+            await redis_conn.rpush(
+                self.redis_list_name,
+                pickle.dumps(message),
+            )
+
+    async def listen(self) -> AsyncGenerator[BrokerMessage, None]:
+        """
+        Listen redis list for new messages.
+
+        This function listens to list and yields new messages.
+
+        :yields: parsed broker messages.
+        """
+        async with Redis(connection_pool=self.connection_pool) as redis_conn:
+            while True:
+                redis_pickled_message = await redis_conn.blpop(
+                    self.redis_list_name,
+                )
+                if redis_pickled_message:
+                    try:  # noqa: WPS448
+                        redis_message = pickle.loads(  # noqa: WPS220
+                            redis_pickled_message[1],
+                        )
+                        yield BrokerMessage(  # noqa: WPS220
```

Comment:
Why do you need another allocation? 
You've already received a python object, so you can yield it, instead of creating a new one broker message.

Like this:
```python
yield pickle.loads(  # noqa: WPS220
    redis_pickled_message[1],
)
```

Reply:
I made it because there may be messages in the queue that were not formed by taskiq, or someone changed the value of an existing message